src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.cpp

```diff
@@ -0,0 +1,100 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp"
+
+#include <array>
+#include <cstddef>
+#include <tuple>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "Domain/Structure/CreateInitialMesh.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Evolution/DiscontinuousGalerkin/MortarData.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/MortarHelpers.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Projection.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace evolution::dg::Initialization {
+template <size_t Dim, bool AddFluxBoundaryConditionMortars>
+auto Mortars<Dim, AddFluxBoundaryConditionMortars>::apply_impl(
+    const std::vector<std::array<size_t, Dim>>& initial_extents,
+    const Element<Dim>& element, const TimeStepId& next_temporal_id,
+    const std::unordered_map<Direction<Dim>, Mesh<Dim - 1>>& interface_meshes,
+    const std::unordered_map<Direction<Dim>, Mesh<Dim - 1>>&
+        boundary_meshes) noexcept
+    -> std::tuple<MortarMap<evolution::dg::MortarData<Dim>>,
+                  MortarMap<Mesh<Dim - 1>>,
+                  MortarMap<std::array<Spectral::MortarSize, Dim - 1>>,
+                  MortarMap<TimeStepId>> {
+  static constexpr size_t dim = Dim;
```

Comment:
In the body of the function you often use Dim and twice use dim. I suggest to remove this and use `Dim` consistently.